kfs-core/src/main/java/edu/arizona/kfs/gl/batch/service/impl/BudgetAdjustmentCashTransferServiceImpl.java

```diff
@@ -0,0 +1,596 @@
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.sql.Date;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.coa.businessobject.Account;
+import org.kuali.kfs.coa.businessobject.ObjectCode;
+import org.kuali.kfs.coa.service.ObjectCodeService;
+import org.kuali.kfs.fp.document.BudgetAdjustmentDocument;
+import org.kuali.kfs.fp.document.validation.impl.BudgetAdjustmentDocumentRuleConstants;
+import org.kuali.kfs.fp.document.validation.impl.TransferOfFundsDocumentRuleConstants;
+import org.kuali.kfs.gl.batch.PosterEntriesStep;
+import org.kuali.kfs.gl.batch.service.RunDateService;
+import org.kuali.kfs.gl.businessobject.OriginEntryFull;
+import org.kuali.kfs.gl.businessobject.OriginEntryInformation;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.businessobject.SystemOptions;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.kfs.sys.Message;
+import org.kuali.kfs.sys.service.OptionsService;
+import org.kuali.kfs.sys.service.ReportWriterService;
+import org.kuali.kfs.sys.service.UniversityDateService;
+import org.kuali.rice.core.api.datetime.DateTimeService;
+import org.kuali.rice.core.api.parameter.ParameterEvaluatorService;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.coreservice.framework.parameter.ParameterService;
+import org.kuali.rice.krad.util.ObjectUtils;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.arizona.kfs.gl.GeneralLedgerConstants;
+import edu.arizona.kfs.gl.batch.service.BudgetAdjustmentCashTransferService;
+import edu.arizona.kfs.gl.businessobject.BudgetAdjustmentTransaction;
+import edu.arizona.kfs.gl.dataaccess.BudgetAdjustmentTransactionDao;
+
+@Transactional
+public class BudgetAdjustmentCashTransferServiceImpl implements BudgetAdjustmentCashTransferService {
+    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(BudgetAdjustmentCashTransferServiceImpl.class);
+
+    protected BudgetAdjustmentTransactionDao budgetAdjustmentTransactionDao;
+    protected ObjectCodeService objectCodeService;
+    protected OptionsService optionsService;
+    protected ParameterService parameterService;
+    protected DateTimeService dateTimeService;
+    protected RunDateService runDateService;    
+    protected ReportWriterService reportWriterService;
+    protected UniversityDateService universityDateService;
+    protected String batchFileDirectoryName;
+        
+    /**
+     * Reads an incoming file of general ledger transactions, extracts those transaction that qualify as budget adjustment transactions 
+     * and saves those transactions to holding table GL_BUDGET_ADJUST_TRN_T.
+     */
+    @Override
+    public void extractAndSaveBudgetAdjustmentEntries() {
+    	     
```

Comment:
I see that we are throwing exceptions on lines 80 and 174 without any message or logging. Let's discuss where else logging might be needed.